src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,130 @@
 import { Button, ButtonGroup, Center, Flex, Stack } from '@chakra-ui/react'
-import { FaChartSimple } from 'react-icons/fa6'
-import { useTranslate } from 'react-polyglot'
-import { RawText } from 'components/Text'
+import { thorchainAssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import type { SingleValueData } from 'lightweight-charts'
+import { useCallback, useMemo, useState } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Interval } from 'react-queries/queries/midgard'
+import { SimpleChart } from 'components/SimpleChart'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import type {
+  MidgardSwapHistoryResponse,
+  MidgardTvlHistoryResponse,
+} from 'lib/utils/thorchain/lp/types'
+import { selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const swapHistoryToChartData = (
+  swapHistory: MidgardSwapHistoryResponse | undefined,
+  runePrice: string,
+): SingleValueData<number>[] => {
+  if (!swapHistory) return []
+
+  return swapHistory.intervals.map(interval => {
+    const intervalVolume2FiatUserCurrency = fromThorBaseUnit(interval.totalVolume)
+      .times(runePrice)
```

Comment:
we will want to use `interval.runePriceUSD` for the calculation and then convert to user fiat currency. volume is static for an interval and not tied to current rune price being passed in

Reply:
678d271bcd83215ae9d245d408a17df85b0495cf